### PR TITLE
Fix #2427: Arxiv fetcher works with prefix

### LINKED ISSUE #2427: arXiv fetcher cannot deal with the "arXiv:" prefix.
JabRef version JabRef 3.8.1-dev--snapshot--2016-12-22--master--6f40cf8
linux 4.8.11-100.fc23.x86_64 amd64 
Java 1.8.0_51 on Fedora 23

Steps to reproduce:

1. Try to fetch infos about an arXiv preprint without removing the "arXiv:" prefix, e.g. "arXiv:0710.0994"
2. A dialog appears for a second, after which nothing happens. The status line still shows "Status: Processing 0710.0994" though.
3. If one removes the prefix, i.e. enters just "0710.0994", then everything works as expected.

In the terminal I don't see any new log messages when this happens.

### PR DESCRIPTION
**Pull request by tobiasdiez**
- [x] Change in CHANGELOG.md described
- [x] Tests created for changes
- [ ] Screenshots added (for bigger UI changes)
- [ ] Manually tested changed features in running JabRef
- [ ] Check documentation status (Issue created for outdated help page at [help.jabref.org](https://github.com/JabRef/help.jabref.org/issues)?)
- [ ] If you changed the localization: Did you run `gradle localizationUpdate`?

### CODE REVIEW

File: src/main/java/net/sf/jabref/logic/importer/fetcher/ArXiv.java [hunk 1]
```diff
@@ -107,6 +107,10 @@ public ArXiv(ImportFormatPreferences importFormatPreferences) {
     }
 
     private Optional<ArXivEntry> searchForEntryById(String identifier) throws FetcherException {
+        identifier = identifier.replace("arxiv:", "");
+        identifier = identifier.replace("arXiv:", "");
```
Comment (stefan-kolb) [inline src/main/java/net/sf/jabref/logic/importer/fetcher/ArXiv.java]: Can you just replace this via a regex? `/arxiv:?/i`

File: src/test/java/net/sf/jabref/logic/importer/fetcher/ArXivTest.java [hunk 2]
```diff
@@ -76,6 +76,14 @@ public void findByEprint() throws IOException {
     }
 
     @Test
+    // Test for https://github.com/JabRef/jabref/issues/2427
```
Comment (stefan-kolb) [inline src/test/java/net/sf/jabref/logic/importer/fetcher/ArXivTest.java]: I would omit the comment here. It is a test for expected behavior. Doesnt matter which bug/problem it fixed.

File: src/test/java/net/sf/jabref/logic/importer/fetcher/ArXivTest.java [hunk 3]
```diff
@@ -139,6 +147,12 @@ public void searchEntryByIdWith4Digits() throws Exception {
...
```
Comment (stefan-kolb) [inline src/test/java/net/sf/jabref/logic/importer/fetcher/ArXivTest.java]: Same as above

File: src/main/java/net/sf/jabref/logic/importer/fetcher/ArXiv.java (hunk 1 above)
Comment (simonharrer) [inline src/main/java/net/sf/jabref/logic/importer/fetcher/ArXiv.java]: I would vote for a ArXiv class, similar to the DOI one which can handle such a case.

File: src/test/java/net/sf/jabref/logic/importer/fetcher/ArXivTest.java [hunk 4]
```diff
@@ -76,6 +76,14 @@ public void findByEprint() throws IOException {
...
+    public void findByEprintWithPrefix() throws IOException {
+        entry.setField("eprint", "arXiv:1603.06570");
+
```
Comment (Siedlerchr) [inline src/test/java/net/sf/jabref/logic/importer/fetcher/ArXivTest.java]: Remove empty line

File: src/test/java/net/sf/jabref/logic/importer/fetcher/ArXivTest.java (hunk 3 above)
Comment (Siedlerchr) [inline src/test/java/net/sf/jabref/logic/importer/fetcher/ArXivTest.java]: Remove url

**Comment by koppor** [comment]
I merged it and will do the requested changes now to get this into 3.8.2.

**Pull request merged**